CMakeLists.txt

```
@@ -426,32 +442,49 @@ if(MINGW)
 endif()
 
 set(lego1_objects)
-set(lego1_link_libraries dxguid d3drm_guid)
+set(lego1_link_libraries)
+if(WIN32)
+  list(APPEND lego1_link_libraries dxguid d3drm_guid)
+endif()
 foreach(lego1_library IN LISTS lego1_targets)
   target_compile_definitions(${lego1_library}-objects PRIVATE LEGO1_DLL)
   list(APPEND lego1_objects $<TARGET_OBJECTS:${lego1_library}-objects>)
   list(APPEND lego1_link_libraries ${lego1_library}-interface)
 endforeach()
-add_library(lego1 SHARED
-  LEGO1/main.cpp
-  ${lego1_objects}
-)
+if (CMAKE_BUILD_TYPE STREQUAL "Release")
+  add_library(lego1 SHARED
+    LEGO1/main.cpp
+    ${lego1_objects}
+  )
+else()
+  add_library(lego1 STATIC
+    LEGO1/main.cpp
+    ${lego1_objects}
+  )
+endif()
 target_link_libraries(lego1 PUBLIC ${lego1_link_libraries})
 
 # Make sure filenames are ALL CAPS
 set_property(TARGET lego1 PROPERTY OUTPUT_NAME LEGO1)
 set_property(TARGET lego1 PROPERTY PREFIX "")
-set_property(TARGET lego1 PROPERTY SUFFIX ".DLL")
+if(WIN32)
+  set_property(TARGET lego1 PROPERTY SUFFIX ".DLL")
+else()
+  set_property(TARGET lego1 PROPERTY SUFFIX ".so")
+  target_compile_options(lego1 PRIVATE -fPIC)
+endif()
```

macOS wants `.dylib` :)
So let's only change Windows.

```suggestion
set_property(TARGET lego1 PROPERTY SUFFIX POSITION_INDEPENDENT_CODE TRUE)
if(WIN32)
  set_property(TARGET lego1 PROPERTY SUFFIX ".DLL")
endif()
```

But again, this can be removed in later work since we don't care about accuracy in this repo.
What's important in this repo is to keep it easy to merge with the other isle repo.

---

Oh is it just there for .dll vs .DLL?

---

Yeah, this is a silly thing.